src/web/rustdoc.rs

```diff
@@ -812,4 +863,297 @@ mod test {
+
+            assert_platform_links(
+                web,
+                "/dummy/0.2.0/dummy/",
+                &[("x86_64-pc-windows-msvc", "/dummy/0.2.0/dummy/index.html")],
+            )?;
+
+            assert_platform_links(
+                web,
+                "/dummy/0.2.0/dummy/index.html",
+                &[("x86_64-pc-windows-msvc", "/dummy/0.2.0/dummy/index.html")],
+            )?;
+
+            assert_platform_links(
+                web,
+                "/dummy/0.2.0/dummy/struct.Dummy.html",
+                &[(
+                    "x86_64-pc-windows-msvc",
+                    "/dummy/0.2.0/dummy/struct.Dummy.html",
+                )],
+            )?;
+
+            // set an explicit target without cross-compile
+            db.fake_release()
+                .name("dummy")
+                .version("0.3.0")
+                .rustdoc_file("dummy/index.html", b"some content")
+                .rustdoc_file("dummy/struct.Dummy.html", b"some other content")
+                .default_target("x86_64-unknown-linux-gnu")
+                .create()?;
+
+            assert_platform_links(
+                web,
+                "/dummy/0.3.0/dummy/",
+                &[("x86_64-unknown-linux-gnu", "/dummy/0.3.0/dummy/index.html")],
+            )?;
+
+            assert_platform_links(
+                web,
+                "/dummy/0.3.0/dummy/index.html",
+                &[("x86_64-unknown-linux-gnu", "/dummy/0.3.0/dummy/index.html")],
+            )?;
+
+            assert_platform_links(
+                web,
+                "/dummy/0.3.0/dummy/struct.Dummy.html",
+                &[(
+                    "x86_64-unknown-linux-gnu",
+                    "/dummy/0.3.0/dummy/struct.Dummy.html",
+                )],
+            )?;
+
+            // multiple targets
+            db.fake_release()
+                .name("dummy")
+                .version("0.4.0")
+                .rustdoc_file("settings.html", b"top-level items")
+                .rustdoc_file("dummy/index.html", b"some content")
+                .rustdoc_file("dummy/struct.Dummy.html", b"some other content")
+                .rustdoc_file("dummy/struct.DefaultOnly.html", b"some otter content")
+                .rustdoc_file("x86_64-pc-windows-msvc/dummy/index.html", b"some content")
+                .rustdoc_file(
+                    "x86_64-pc-windows-msvc/dummy/struct.Dummy.html",
+                    b"some other content",
+                )
+                .rustdoc_file(
+                    "x86_64-pc-windows-msvc/dummy/struct.WindowsOnly.html",
+                    b"some otter content",
+                )
+                .default_target("x86_64-unknown-linux-gnu")
+                .add_target("x86_64-pc-windows-msvc")
+                .create()?;
+
+            // For top-level items on non-default platforms we redirect to the target-specific doc
+            // root as the top-level items can't know which target they're for
```

Comment:
This is actually a bad assumption, `settings.html` is distributed for each target.

Reply:
There is no need to have one `settings.html` per target. Per rustdoc version eventually but that's it.

Reply:
Hmm, maybe we should add it to `ESSENTIAL_FILES` with everything else that's shared between rustdoc versions?

Reply:
But either way it doesn't need to be addressed in this PR.